packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx

```diff
@@ -0,0 +1,107 @@
+
+```typescript!
+useClientEffect$(()=> { 
+    const listener = (event)=> { 
+        const mouseEvent = event as MouseEvent
+        console.log(mouseEvent.x, mouseEvent.y);
+    }
+    document.addEventListener('mousemove', listener);
+    
+    return ()=> {document.removeEventListener(listener)}
+})
+```
+
+**Why is above a problem**:
+This forces more javascript to eagerly load, rather than be surgically reactive to user events. More upfront javascript => slower app performance. 
+
+It also requires you, the developer, to stay more aware of the server/client relationship (there's a reason you reached for `use >Client< Effect` after all). 
+
+While that's sometimes necessary, it should be the job of the framework to keep you from having to worry about it as much as possible.
+
+Not to mention it requires the extra effort for you to manually clean up the listener. 
+
+**Do this instead**:
+
+```typescript!
+useOnDocument('mousemove', $((event)=> { 
+    const mouseEvent = event as MouseEvent;
+    console.log(mouseEvent.x, mouseEvent.y)
+    // No manual clean up required!
+}))
+```
+
+> NOTE: Similar hooks exist for window and individual components.  
+> `window` -> `useOnWindow` , component -> `useOn` 
+> * [useOn / useDocument / useWindow docs](https://qwik.builder.io/tutorial/hooks/use-on/)
+> * [Example](https://qwik.builder.io/tutorial/hooks/use-on/#example)
+> * [Tutorial](https://qwik.builder.io/tutorial/events/programmatic/)
+----
+
+
+
+## Don't rely on `window` to get the location
+
+Instead => `useLocation()`
+
+**anti-pattern**:
+
+```typescript!
+useClientEffect$(()=> { 
+    if (window.location.href).includes('foo') { 
+        //... do the thing
+    }
+})
+
+//OR 
+
+//(in the root of a component)
+if (typeof window !== "undefined") { 
+    const queryParams = new URLSearchParams(window.location.search);
+    const query: Record<string, string> = {};
+    queryParams.forEach((value, key) => { 
+        query[key] = value;
+    })
+    doTheThing(query);
+}
+```
+
+**Why is above a problem**:
+Many, many things you'll do in response to location information can easily be done on the initial render on the server, and will result in pure HTML with no JS overhead. 
+
+If you force that logic to instead run on the client side, it introduces more upfront JS & causes an eager load. 
+
+
+When using the `if typeof window !== "undefined"` pattern you'll soon find it will simply skip that code. It will run one time on the server and skip the code block because window will always be undefined on the server. You're used to code running twice. Qwik doesn't need to! 😎
+
+**Do this instead**:
+```typescript!
+const location = useLocation();
+
+if (location.href.includes('foo')) { 
+    // Do the thing
+}
+//OR 
+doTheThing(location.query);
+    
+```
+
+
+## EXCEPTION
+When doing SSG for pure static files, this is a necessary evil. The server won't have current location information when the server generates the files at build time. 
```

Comment:
I don't understand this. `useLocation()` gets the location of the page that is being SSR'ed, no? And then when it runs on the client it gets the actual file location? Doesn't it still work with SSG?

Reply:
@KenAKAFrosty

Reply:
yeah, i dont understand this exception either, `useLocation()` will also work in SSG, SSG is the same as SSR (just cached forever)

Reply:
@mhevery